### PR TITLE
[docs] - Condense & clean up 'Getting help'

### PR DESCRIPTION
**Pull request by erinkcochran87**
## Summary & Motivation

This PR updates the content in the **Getting help** docs page:

- To be more visually concise
- Add a summary-level table of resources and when to use them
- Create a dedicated **Tips** section with info from various sections

Putting myself in a user's shoes, I wouldn't want to read a lengthy page that tells me how to get help if I'm already struggling. I'd want something skimmable and to the point. The resource summary table frames different resources as what they can help users do, versus having to read a lot of copy and then figure it out.

### New version

Screenshot 2024-03-19 at 7 37 25 PM

Screenshot 2024-03-19 at 7 38 38 PM

### Previous version

Screenshot 2024-03-19 at 7 37 35 PM

---

## How I Tested These Changes

👀

### CODE REVIEW

File: docs/content/getting-started/getting-help.mdx [hunk 1]
```diff
@@ -1,84 +1,271 @@
...
+### Ask questions that get answers
+
+When posting questions in [GitHub Discussions](https://github.com/dagster-io/dagster/discussions) or [Slack](https://dagster.io/slack), we recommend:
+
+- Being clear, specific, and as brief as possible
+- Not including lots of irrelevant details
+
+Generally, the less time and effort it takes for someone to digest your question, the more likely it will get answered. Check out the following table for some examples of well-phrased questions and those that could use some improvement.
+
+<table
```
Comment (erinkcochran87) [inline docs/content/getting-started/getting-help.mdx:161]: I'm kind of on the fence about how these questions are formatted, but wanted feedback.

Comment (yuhan) [reply docs/content/getting-started/getting-help.mdx:161]: it took a while for me to internalize what qualifies a good question - i feel it's a lot going on with the bold and italic fonts.

how about this? https://gist.github.com/yuhan/fbd8164abfccef62ee0c750a215a1e5f

File: docs/content/getting-started/getting-help.mdx [hunk 2]
```diff
@@ -1,84 +1,271 @@
...
-## Searching for answers
-
-Search engines like Google generally do a good job at surfacing relevant Dagster docs, Github Discussions and Github Issues, so we recommend starting there.
-
-Unfortunately, ChatGPT is not currently a great resource for answering questions about Dagster, because the corpus it was trained on doesn’t include Dagster’s latest APIs.
-
-However, one can interact with a fine-tuned Language Learning Model (LLM) trained on the latest Dagster docs, Github Discussions and Github Issues right here within our docs pages! Simply click the **Ask AI** button on any docs page to open a chat window to it.
-
-Our LLM is also available in the [#ask-ai](https://dagster.slack.com/archives/C066HKS7EG1) channel of the Dagster Slack.
+## Available resources
```
Comment (yuhan) [inline docs/content/getting-started/getting-help.mdx:12]: i like this!

File: docs/content/getting-started/getting-help.mdx [hunk 3]
```diff
@@ -1,84 +1,271 @@
...
+  </tbody>
+</table>
 
 ---
 
-## Bugs and feature requests
+## Tips
 
-We use [Github Issues](https://github.com/dagster-io/dagster/issues) to track all bugs and feature requests from the community, and we welcome issues submitted by users.
+### Code snippets
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/getting-started/getting-help.mdx:121]: > When posting in GitHub or Slack, including a code snippet can help others understand your question. Using Markdown to format the content as code improves readability. For example:

?

File: docs/content/getting-started/getting-help.mdx (hunk 1 above)
Comment (erinkcochran87) [reply docs/content/getting-started/getting-help.mdx:161]: I dig this, thank you!

File: docs/content/getting-started/getting-help.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuhan) [reply docs/content/getting-started/getting-help.mdx:121]: oh i was more referring to just the headings as it's not intuitive to me from the headings whether this applies to other available resources (online search and ask ai) vs just GH. 
image

File: docs/content/getting-started/getting-help.mdx [hunk 4]
```diff
@@ -1,84 +1,197 @@
...
+When filing a [GitHub issue](https://github.com/dagster-io/dagster/issues) for a bug:
 
-If you're trying to find out the best way to use Dagster for your use case, there are a couple places to get help.
+- **DO** include the minimum-sized working code snippet that reproduces the issue
+- **DO** include the steps you took to create the issue, if the issue occurred in the Dagster UI
+- **DO** include the Dagster version being used, if applicable
+- **DON'T** include any sensitive information
 
-### Github Discussions: the preferred place to ask questions
+### Ask questions that get answers
```
Comment (yuhan) [inline docs/content/getting-started/getting-help.mdx:150]: actually, do we want to promote this to the top? i feel this is actually the most important tip.

File: docs/content/getting-started/getting-help.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/content/getting-started/getting-help.mdx:121]: It's not all GH though - it applies to Slack as well. Do you feel strongly that it needs to be that spelled out?

File: docs/content/getting-started/getting-help.mdx (hunk 4 above)
Comment (yuhan) [reply docs/content/getting-started/getting-help.mdx]: no i dont feel strongly. your call!

File: docs/content/getting-started/getting-help.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuhan) [reply docs/content/getting-started/getting-help.mdx]: oh i see you called out the specific language is for GH (i was under the impression that this is for GH) ignore my comment then. i think it makes sense now.

**Review by yuhan — APPROVED** [review]

**Pull request merged**